src/ast/mod.rs

```diff
@@ -733,6 +733,10 @@ pub enum Expr {
     ///
     /// See <https://docs.snowflake.com/en/sql-reference/constructs/where#joins-in-the-where-clause>.
     OuterJoin(Box<Expr>),
+    ColumnWithDataType {
```

Comment:
Could we include a description for this new variant (ideally including a link to the docs)?

Reply:
Sure!